lib/app_web/controllers/google_auth_controller.ex

```diff
@@ -0,0 +1,83 @@
+defmodule AppWeb.GoogleAuthController do
+  use AppWeb, :controller
+  # use AppWeb, :router
+  alias AppWeb.Router.Helpers
+  alias App.Ctx.Session
+  # alias App.Repo
+
+  def index(conn, %{"code" => code}) do
+    {:ok, token} = ElixirAuthGoogle.get_token(code)
+    {:ok, profile} = ElixirAuthGoogle.get_user_profile(token["access_token"])
+    # IO.inspect profile, label: "profile"
+    person = AppWeb.GoogleAuthController.transform_profile_data_to_person(profile)
+
+    # get the person by email
+    case App.Ctx.get_person_by_email(profile["email"]) do
+      nil ->
+        # Create the person
+        {:ok, google_person} = App.Ctx.create_google_person(person)
+
+        # Create session
+        session_attrs = %{
+          "auth_token" => token["access_token"],
+          "refresh_token" => token["refresh_token"]
+        }
+
+        App.Ctx.create_session(google_person, session_attrs)
+
+      person ->
+        # create new session and
+        session_attrs = %{
+          "auth_token" => token["access_token"],
+          "refresh_token" => "dummy_refresh_token", # we don't need refresh for now.
+        }
+
+        App.Ctx.create_session(person, session_attrs)
+    end
+
+    # redirect(conn, to: "/")
+    # IO.inspect person, label: "person"
+    render(conn, "index.html", person: person)
+  end
+
+  @doc """
+  `transform_profile_data_to_person/1` transforms the profile data
+  received from invoking `ElixirAuthGoogle.get_user_profile/1`
+  into a `person` record that can be inserted into the people table.
+
+  ## Example
+
+    iex> transform_profile_data_to_person(%{
+      "email" => "nelson@gmail.com",
+      "email_verified" => true,
+      "family_name" => "Correia",
+      "given_name" => "Nelson",
+      "locale" => "en",
+      "name" => "Nelson Correia",
+      "picture" => "https://lh3.googleusercontent.com/a-/AAuE7mApnYb260YC1JY7a",
+      "sub" => "940732358705212133793"
+    })
+    %{
+      "email" => "nelson@gmail.com",
+      "email_verified" => true,
+      "family_name" => "Correia",
+      "given_name" => "Nelson",
+      "locale" => "en",
+      "name" => "Nelson Correia",
+      "picture" => "https://lh3.googleusercontent.com/a-/AAuE7mApnYb260YC1JY7a",
+      "sub" => "940732358705212133793"
+      "status" => 1,
+      "familyName" => "Correia",
+      "givenName" => "Nelson"
+    }
+  """
+  def transform_profile_data_to_person(profile) do
```

Comment:
I think this function can be defined in the `Person` context as a specific changeset function.

Reply:
@SimonLab indeed. please feel free to move it.